utils/componenterror/componenterror.go

```diff
@@ -0,0 +1,78 @@
+// Copyright AGNTCY Contributors (https://github.com/agntcy)
+// SPDX-License-Identifier: Apache-2.0
+
+package componenterror
+
+import (
+	"errors"
+	"maps"
+	"strings"
+
+	"google.golang.org/genproto/googleapis/rpc/errdetails"
+)
+
+type ComponentError struct {
+	Err error `json:"-"`
+
+	Component string            `json:"component"`
+	Metadata  map[string]string `json:"metadata,omitempty"`
+	Message   string            `json:"message"`
+}
+
+func NewComponentError(err error, component string) *ComponentError {
+	cErr := &ComponentError{
+		Err:       err,
+		Component: component,
+	}
+
+	if err != nil {
+		cErr.Message = err.Error()
+	}
+
+	return cErr
+}
+
+func (e *ComponentError) Error() string {
+	return e.Message
```

Comment:
i would opt to add a TODO note here

```suggestion
	// TODO: decide how we want to format the component error message in the future, e.g. do we want to include annotations or not
	return e.Message
```